src/parser/mod.rs

```diff
@@ -1222,6 +1222,15 @@ impl<'a> Parser<'a> {
             Token::Mul => {
                 return Ok(Expr::Wildcard(AttachedToken(next_token)));
             }
+            // Handle parenthesized wildcard: (*)
+            Token::LParen => {
+                let inner_token = self.next_token();
+                if inner_token.token == Token::Mul && self.peek_token().token == Token::RParen {
+                    self.next_token(); // consume RParen
+                    return Ok(Expr::Wildcard(AttachedToken(inner_token)));
+                }
+                // Not a (*), reset and fall through to parse_expr
```

Comment:
this looks incorrect per the comment, by reset should we have called `prev_token()` or similar to undo the consumed inner token `*`?